components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/CacheBackedWebhookManagementDAO.java

```diff
@@ -0,0 +1,149 @@
+
+    private static final Log LOG = LogFactory.getLog(CacheBackedWebhookManagementDAO.class);
+    private final WebhookManagementDAO webhookManagementDAO;
+    private final WebhookCache webhookCache;
+
+    /**
+     * Constructor.
+     *
+     * @param webhookManagementDAO WebhookManagementDAO implementation to be wrapped.
+     */
+    public CacheBackedWebhookManagementDAO(WebhookManagementDAO webhookManagementDAO) {
+
+        this.webhookManagementDAO = webhookManagementDAO;
+        this.webhookCache = WebhookCache.getInstance();
+    }
+
+    @Override
+    public void createWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        webhookManagementDAO.createWebhook(webhook, tenantId);
+    }
+
+    @Override
+    public Webhook getWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+
+        WebhookCacheEntry webhookCacheEntry = webhookCache.getValueFromCache(new WebhookCacheKey(webhookId), tenantId);
+        if (webhookCacheEntry != null && webhookCacheEntry.getWebhook() != null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Webhook cache hit for webhook ID: " + webhookId + ". Returning from cache.");
+            }
+            return webhookCacheEntry.getWebhook();
+        }
+
+        Webhook webhook = webhookManagementDAO.getWebhook(webhookId, tenantId);
+        if (webhook != null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Webhook cache miss for webhook ID: " + webhookId + ". Adding to cache.");
+            }
+            webhookCache.addToCache(new WebhookCacheKey(webhookId), new WebhookCacheEntry(webhook), tenantId);
+        }
+        return webhook;
+    }
+
+    @Override
+    public List<String> getWebhookEvents(String webhookId, int tenantId) throws WebhookMgtException {
+
+        WebhookCacheEntry webhookCacheEntry = webhookCache.getValueFromCache(new WebhookCacheKey(webhookId), tenantId);
+        if (webhookCacheEntry != null && webhookCacheEntry.getWebhook() != null &&
+                webhookCacheEntry.getWebhook().getEventsSubscribed() != null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Webhook cache hit for webhook ID: " + webhookId + ". Returning from cache.");
+            }
+            return webhookCacheEntry.getWebhook().getEventsSubscribed();
+        }
+
+        Webhook webhook = webhookManagementDAO.getWebhook(webhookId, tenantId);
+        if (webhook != null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Webhook cache miss for webhook events for webhook ID: " + webhookId + ". Adding to cache.");
+            }
+            webhookCache.addToCache(new WebhookCacheKey(webhookId), new WebhookCacheEntry(webhook), tenantId);
+
+            if (webhook.getEventsSubscribed() != null) {
+                return webhook.getEventsSubscribed();
+            }
+        }
+        return java.util.Collections.emptyList();
```

Comment:
why not have this imported so in the code it's just Collections.emptyList()

Reply:
fixed